### PR TITLE
cck: add cases for pending, skipped and undefined

### PR DESCRIPTION
**Pull request by davidjgoss**
## Summary

Add cases to the compatibility kit for:

- Pending steps
- Skipped scenarios via hooks/steps
- Undefined steps

## Motivation and Context

We want to document and standardise the behaviour of these features. We'll also need them in order to use the CCK as the reference point for a JUnit formatter (see #871).

## How Has This Been Tested?

Test added to fake-cucumber which now supports `skipped`.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] The change has been ported to Java.
- [x] The change has been ported to Ruby.
- [ ] The change has been ported to JavaScript.
- [ ] The change has been ported to Go.
- [ ] The change has been ported to .NET.
- [x] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have updated the CHANGELOG accordingly.

### CODE REVIEW

File: compatibility-kit/javascript/features/pending/pending.feature [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  not yet implemented (or "pending") by returning or throwing a particular
+  value.
+
+  This causes subsequent steps in the scenario to be skipped, and the overall
+  result to be treated as a failure.
+
+  Scenario: Unimplemented step signals pending status
+    Given a step
+    When a step that isnt implemented yet
+    Then a step
```
Comment (mpkorstanje) [inline compatibility-kit/javascript/features/pending/pending.feature:13]: Dropping the other two steps would simply this.

File: compatibility-kit/javascript/features/skipped/skipped.feature [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  for running the scenario.
+
+  @skip
+  Scenario: Skipping from a Before hook
+    Given a step
+
+  Scenario: Skipping from a step
+    Given a step
+    When a step that skips
+    Then a step
```
Comment (mpkorstanje) [inline compatibility-kit/javascript/features/skipped/skipped.feature:16]: Same as before. There are a few more.

File: compatibility-kit/javascript/features/pending/pending.feature (hunk 1 above)
Comment (davidjgoss) [reply compatibility-kit/javascript/features/pending/pending.feature:13]: I wanted the messages that come out of this to show that the step before runs and passes, and the step after is skipped - I think that's key to properly implementing this?

Comment (davidjgoss) [reply compatibility-kit/javascript/features/pending/pending.feature:13]: Maybe I should make this explicit in the wording of these steps?

Comment (mpkorstanje) [reply compatibility-kit/javascript/features/pending/pending.feature:13]: > I wanted the messages that come out of this to show that the step before runs and passes, and the step after is skipped - I think that's key to properly implementing this?

I think that should be a separate test case then. Could be much more generic. Any step after a non-passing step is skipped.

Comment (davidjgoss) [reply compatibility-kit/javascript/features/pending/pending.feature:13]: I kind of prefer to have the test cases for pending etc be fairly complete even if there is a little duplication overall with the skip-subsequent-steps concept, especially as there is some patchiness in our implementations (e.g. ruby doesn't support pending or skipped it seems).

Comment (mpkorstanje) [reply compatibility-kit/javascript/features/pending/pending.feature:13]: I understand, but when the actual and expected result differ it is very hard to to see what the problem is when there is extra fluff laying around. 

Consider instead using multiple examples:

```
  Scenario: Unimplemented step signals pending status
    Given a step that isnt implemented yet

  Scenario: Steps after unimplemented steps are skipped
    Given a step that isnt implemented yet
    Then a skipped step

  Scenario: Steps before unimplemented steps are executed
    Given an implemented step
    Then a step that isnt implemented yet
```

Comment (davidjgoss) [reply compatibility-kit/javascript/features/pending/pending.feature]: Thanks - I've refactored based on your suggestion above.

**Review by aurelien-reeves — APPROVED** [review]
Looks good to me

**Pull request merged**